core-web/libs/edit-content/src/lib/fields/dot-edit-content-category-field/components/dot-category-field-search-list/dot-category-field-search-list.component.ts

```diff
@@ -0,0 +1,177 @@
+import { DotTableSkeletonComponent } from '../dot-table-skeleton/dot-table-skeleton.component';
+
+const DELAY_FOR_LISTENER = 300;
+
+@Component({
+    selector: 'dot-category-field-search-list',
+    standalone: true,
+    imports: [CommonModule, TableModule, SkeletonModule, DotTableSkeletonComponent, DotMessagePipe],
+    templateUrl: './dot-category-field-search-list.component.html',
+    styleUrl: './dot-category-field-search-list.component.scss',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotCategoryFieldSearchListComponent implements AfterViewInit {
+    /**
+     * Represents a reference to a table container element in the DOM to calculate the
+     * viewport to use in the virtual scroll
+     */
+    @ViewChild('tableContainer', { static: false }) tableContainer!: ElementRef;
+
+    /**
+     * The scrollHeight variable represents a signal with a default value of '0px'.
+     * It can be used to track and manipulate the height of a scrollable element.
+     */
+    scrollHeight = signal<string>('0px');
+
+    /**
+     * Represents the categories found with the filter
+     */
+    categories = input.required<DotCategoryFieldCategory[]>();
+
+    /**
+     * Represent the selected items in the store
+     */
+    selected = input.required<DotCategoryFieldKeyValueObj[]>();
+
+    /**
+     * EventEmitter for emit the selected category(ies).
+     */
+    @Output() itemChecked = new EventEmitter<
+        DotCategoryFieldKeyValueObj | DotCategoryFieldKeyValueObj[]
+    >();
+
+    /**
+     * EventEmitter that emits events to remove a selected item(s).
+     */
+    @Output() removeItem = new EventEmitter<string | string[]>();
+
+    /**
+     * Represents a variable indicating if the component is in loading state.
+     */
+    isLoading = input.required<boolean>();
+
+    /**
+     * Computed variable to store the search results parsed.
+     *
+     */
+    searchResults = computed<DotCategoryFieldKeyValueObj[]>(() => {
+        return this.categories().map((item) => {
+            const path = getParentPath(item);
+
+            return { key: item.key, value: item.categoryName, path: path, inode: item.inode };
+        });
+    });
+
+    /**
+     * Model of the items selected
+     */
+    itemsSelected: DotCategoryFieldKeyValueObj[];
+
+    /**
+     * Represents an array of temporary selected items.
+     */
+    temporarySelectedAll: string[] = [];
+
+    #destroyRef = inject(DestroyRef);
+
+    readonly #effectRef = effect(() => {
+        // Todo: find a better way to update this
+        this.itemsSelected = this.selected();
+    });
+
+    private resizeSubject = new Subject();
+
+    ngAfterViewInit(): void {
+        this.resizeSubject
+            .pipe(takeUntilDestroyed(this.#destroyRef), debounceTime(DELAY_FOR_LISTENER))
+            .subscribe(() => this.calculateScrollHeight());
+
+        window.addEventListener('resize', () => this.resizeSubject.next());
```

Comment:
@oidacra avoid using widows directly, I suggest using `@HostListener`, this is the Angular way, here is an example of your approach:

```ts
export class HostListenerComponent { readonly #resizeSubject
  readonly #resizeSubject = new Subject();

  @HostListener('window:resize', ['$event'])
  resize(event: Event) {
    this.#resizeSubject.next(event);
  }

  constructor() {
    this.#resizeSubject
    .pipe( debounceTime(500))
    .subscribe(() => console.log('calc'))
  }

}
```

This also avoids using `ngAfterViewInit` and the listener is automatically removed when the component is destroyed.

Reply:
this only is triggered by `window:resize` I need the container of the table in the component.